tests/Unit/CallPowerTest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Tests\Unit;
+
+use Exception;
+use Tests\TestCase;
+use Chompy\Services\Rogue;
+
+class CallPowerTest extends TestCase
+{
+    /**
+     * Test that a post with a completed status successfully sends to Rogue.
+     *
+     * @return void
+     */
+    public function testCompletedCallStatus()
+    {
+        // Create a post with completed as the status.
+        $response = $this->withChompyApiKey()->json('POST', 'api/v1/callpower/call', [
+            'mobile' => '1234567891',
+            'callpower_campaign_id' => '1',
+            'status' => 'completed',
+            'call_timestamp' => '2017-11-09 06:34:01.185035',
+            'call_duration' => 50,
+            'campaign_target_name' => 'Mickey Mouse',
+            'campaign_target_title' => 'Representative',
+            'campaign_target_district' => 'FL-7',
+            'callpower_campaign_name' => 'Test',
+            'number_dialed_into' => '+12028519273',
+        ]);
+
+        $response->assertStatus(200);
+    }
+
+    /**
+     * Test that a post with a busy status successfully sends to Rogue.
+     *
+     * @return void
+     */
+    public function testBusyCallStatus()
+    {
+        // Create a post with completed as the status.
+        $response = $this->withChompyApiKey()->json('POST', 'api/v1/callpower/call', [
+            'mobile' => '1234567891',
+            'callpower_campaign_id' => '1',
+            'status' => 'busy',
+            'call_timestamp' => '2017-11-09 06:34:01.185035',
+            'call_duration' => 50,
+            'campaign_target_name' => 'Mickey Mouse',
+            'campaign_target_title' => 'Representative',
+            'campaign_target_district' => 'FL-7',
+            'callpower_campaign_name' => 'Test',
+            'number_dialed_into' => '+12028519273',
+        ]);
+
+        $response->assertStatus(200);
+    }
+
+    /**
+     * Test that a post with an invalid CallPower campaign id is not made,
+     * and enters into the failed_jobs table.
+     *
+     * @return void
+     */
+    public function testFailedPostWithInvalidCallPowerCampaignId()
+    {
+        // Mock the Rogue API call to get the action id.
+        $this->mock(Rogue::class)
+            ->shouldNotReceive('getActionIdFromCallPowerCampaignId')->andReturn(new Exception(500))
```

Comment:
I also don't think this is the way to test failed jobs the more I think about it and the more research I do... should this be testing if it makes it into the queued job instead? Maybe this test (to make sure it makes it in the `failed_jobs` table) is not meant for unit test functionality since it is more of an overall "feature" and too big for this type of testing?

Reply:
Yeah, I agree! I think we can trust that Laravel handles putting failed jobs into the `failed_jobs` table since that's built in to the framework. We're just interested in whether the job succeeds or fails for the purposes of this test.

Reply:
ok awesome, thanks! I'll remove this :)